### ISSUE TITLE
Runtime Error: th_index_select not supported

### ISSUE THREAD

**Issue by aastha19** [pytorch/fairseq#1350]
Vocab size= 3445496
Training set size = 26392225

On trying to train a new model using the command:

python train.py data-bin-2cr/ --save-dir checkpoints/trans_lev/ --ddp-backend=no_c10d --task translation_lev --criterion nat_loss --arch levenshtein_transformer --noise random_delete --share-all-embeddings --optimizer adam --adam-betas '(0.9,0.98)' --lr 0.0005 --lr-scheduler inverse_sqrt --min-lr '1e-09' --warmup-updates 10000 --warmup-init-lr '1e-07' --label-smoothing 0.1 --dropout 0.3 --weight-decay 0.01 --decoder-learned-pos --apply-bert-init --log-format 'simple' --log-interval 100 --fixed-validation-seed 7 --max-tokens 8000 --save-interval-updates 10000 --max-update 300000

Following runtime error is seen:
![image](https://user-images.githubusercontent.com/14852345/68286378-044e4c00-00a7-11ea-8970-800a9c0fb4df.png)

Any help would be appreciated!

**Comment by myleott** [comment]
This code is broken currently due to some recent commits. Working on fixing it now.

**Comment by aastha19** [comment]
ok. any idea by when can it be fixed?

**Comment by gvskalyan** [comment]
@aastha19 Can you please tell the datasets which you used for training English-Hindi. If possible please share the stats / BLEU scores on using different architectures and hyperparameters.

**Comment by sdll** [comment]
@myleott, any updates on this?

**Comment by aastha19** [comment]
> @aastha19 Can you please tell the datasets which you used for training English-Hindi. If possible please share the stats / BLEU scores on using different architectures and hyperparameters.

@gvskalyan the data set used is not a publically available dataset. It was made from scratch.
I have tried multiple combinations of architectures and hyperparameters, but the maximum BLEU score on my test data was around 58.86 using LSTM architecture.

**Comment by myleott** [comment]
I think this should be fixed now, can you please try again?

**Comment by aastha19** [comment]
@myleott the training set is loaded and then it just freezes. The training doesn't begin.

![image](https://user-images.githubusercontent.com/14852345/68828914-3e35d880-06cd-11ea-9bfb-4d3fff2541ee.png)

**Comment by MultiPath** [comment]
can you try smaller batch-size and have a check again?

**Comment by aastha19** [comment]
@MultiPath Thanks a lot. 

Reducing the max-tokens to 100 worked. Although when using other architectures like a transformer, LSTM etc. on the same data; max-token size of 3000 or even more would work.

What is so different in this case that it had to be so drastically reduced.
